internal/controller/children.go

```diff
@@ -101,14 +102,34 @@ func (c ChildResource[T]) Reconcile(ctx context.Context, clnt client.Client, cr
 			if !ok {
 				return fmt.Errorf("Could not cast when reconciling")
 			}
+			events := v1.EventList{}
+			selector, err := fields.ParseSelector(fmt.Sprintf("involvedObject.name=%s,involvedObject.namespace=%s,involvedObject.kind=StatefulSet", current.ObjectMeta.Name, current.ObjectMeta.Namespace))
+			if err == nil {
+				if err := clnt.List(ctx, &events, &client.ListOptions{FieldSelector: selector}); err == nil {
+					for _, event := range events.Items {
+						if event.Reason == "FailedCreate" && strings.Contains(event.Message, "exceeded quota") {
+							statusCallback = func(status *amaltheadevv1alpha1.AmaltheaSessionStatus) {
+								status.State = amaltheadevv1alpha1.Failed
+								status.Error = "Quota exceeded"
+							}
+							return nil
+						}
+					}
+				} else {
+					log.Error(err, "couldn't list events")
+
+				}
+			} else {
+				log.Error(err, "couldn't parse selector")
+			}
```

Comment:
suggestion: You can use `client.MatchingFields{}` which can be constructed like a Set and it cannot error out. Rather than parse the selector and then check for error.

See the end of the example in: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#FieldIndexer